### PR TITLE
[cli] Pipelet knactor integration

### PR DESCRIPTION
**Pull request by TarasPriadka**
#### What type of PR is this?
feature

#### What this PR does / why we need it:
Integrate digi library with pipelet knactor with an option to keep old behavior.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
```release-note
- Adding a new command `digi space start pipelet` to start pipelet worker.
- Adding a new option `offload_sync` to egress and ingress fields in the model specifications. If set to true, digis will create pipelet API objects and use pipetor to sync data across pools.
```

### CODE REVIEW

File: space/pipelet/model.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+group: digi.dev
```
Comment (silveryfu) [inline space/pipelet/model.yaml:1]: Let's put the group under `knactor.io`?

**Comment by silveryfu** [comment]
Btw., since the Pipetor isn't maintained in this repo, could you run `build -a` when building the distribution so that people from all arch can access it (e.g. in our GCP dSpaces).

File: space/pipelet/deploy/pipelet/cr.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: digi.dev/v1
+kind: Pipelet
+metadata:
+  name: "p1"
+spec:
```
[the remark this answers is not in the document]
Comment (silveryfu) [reply space/pipelet/deploy/pipelet/cr.yaml:5]: In fact, perhaps we should use this file for the purpose of "[space/pipelet/deploy/pipelet/cr.yaml](https://github.com/digi-project/digi/pull/73/files/bbf4f24750630824e3301f1bcf305e7ebdc63ff3#diff-0a4ee8ed14d818763226f3230b1529427a3b7b17e8dd4f9e2d6dd719b8a2c348)" and then for testing, put it under test/cr.yaml?

File: model/gen.py [hunk 3]
```diff
@@ -168,6 +168,8 @@
     type: boolean
   use_sourcer:
     type: boolean
+  offload_sync:
```
Comment (silveryfu) [inline model/gen.py:171]: @TarasPriadka let's just call this flag `offload`? Then in the code, name the variable `pipelet_offload`

File: driver/digi/data/router.py [hunk 4]
```diff
@@ -68,18 +116,34 @@ def update(self, config: dict):
...
+            if offload_sync:
+                create_custom_api_object(
+                    name=f"{digi.n}-ingress-{name}".replace("_", "-"),
+                    src=sources,
+                    dst=[digi.pool.name],
+                    in_flow=flow,
+                    out_flow=_out_flow,
+                    interval=-1,
+                    eoio=ig.get("eoio", True),
+                    action="create"
```
Comment (silveryfu) [inline driver/digi/data/router.py:129]: Add a TBD for `skip_history` feature?

File: model/Makefile [hunk 5]
```diff
@@ -246,6 +246,12 @@ start-sourcer:
 stop-sourcer:
 	digi stop sourcer
 
+.PHONY: start-pipelet stop-pipelet
+start-pipelet:
+	WORKDIR=$(DIGIHOME) digi run space/pipelet pipetor -d $(CR)
```
Comment (silveryfu) [inline model/Makefile:251]: Name the digi pipetor -> pipelet (I know there is an ambiguity but let's think of it as "using the resource's name to refer to the controller")

File: space/pipelet/deploy/pipelet/cr.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (TarasPriadka) [reply space/pipelet/deploy/pipelet/cr.yaml:5]: Actually, we never really use this CR for pipelet deployment. 
For the purpose of creation of pipelets in k8s datastore we use this one `driver/digi/data/pipelet.yaml`. I had this pipelet CR here for reference in the past during dev, but I think we can remove this

**Comment by TarasPriadka** [comment]
> Btw., since the Pipetor isn't maintained in this repo, could you run `build -a` when building the distribution so that people from all arch can access it (e.g. in our GCP dSpaces).

I am not sure what you mean, the pipetor image will live on Docker Hub right? So they should just be able to pull it when running `digi space start pipelet`, like you can do in `mock` and `example` repos.

File: driver/digi/data/router.py [hunk 6]
```diff
@@ -1,17 +1,63 @@
...
+
 """
 A router contains a collection of pipelets organized as ingresses and egresses.
 Each pipelet is implemented as a digi.data.sync.Sync object that copies and ETL
-data between a source data pool and a destination data pool. 
+data between a source data pool and a destination data pool.
 """
 
 
+def create_custom_api_object(name, src, dst, in_flow, out_flow, interval, eoio, action="create"):
```
Comment (silveryfu) [inline driver/digi/data/router.py:23]: A nit and could be in future PR (let me know your preference :)) - might want to refactor this into a separate `pipelet.py` that takes `pipelet_offload` and spec for pipelet and demux/decide to use Pipetor or local sync? @TarasPriadka

**Comment by silveryfu** [comment]
Lgtm! Great work @TarasPriadka. 

Left one more comment about a slight refactoring -- let me know how you want to proceed.

File: driver/digi/data/router.py (hunk 6 above)
Comment (TarasPriadka) [reply driver/digi/data/router.py:23]: Sure, this sounds like a great idea! I think I will do this in another PR after this one goes in

**Comment by TarasPriadka** [comment]
> Lgtm! Great work @TarasPriadka.
> 
> Left one more comment about a slight refactoring -- let me know how you want to proceed.

Lets merge this PR and I will make another for the refactoring

**Pull request merged**